src/containers/Application.js

```diff
@@ -74,46 +78,73 @@ class Application extends Component {
 	}
 
 	render() {
+		//const authn = useSelector(state => state.authn);
+		console.log("this.Store",this.Store)
+		console.log("store state", this.Store.getState())
 		return (
-			<Provider store={this.Store}>
-			<div className="app">
-				 <AppHeader fixed>
-					<Header app={this}/>
-				</AppHeader>
-				<div className="app-body">
-					<AppSidebar fixed display="lg">
-						<AppSidebarNav navConfig={this.Navigation.getItems()} {...this.props} />
-						<AppSidebarFooter />
-						<AppSidebarMinimizer />
-					</AppSidebar>
-					<main className="main">
-						<AppBreadcrumb appRoutes={this.Router.Routes}/>
+			<Provider store={this.Store}>							
+					<div className="app">
 						<Switch>
 							{this.Router.Routes.map((route, idx) => {
-								return route.component ? (
-									<Route
-										key={idx}
-										path={`${route.path}`}
-										exact={route.exact}
-										name={route.name}
-										render={props => (
-											<route.component {...props} />
-										)} />
-									
+								return route.component ? (							
+										<Route
+											key={idx}
+											path={`${route.path}`}
+											exact={route.exact}
+											name={route.name}
+											render={props => (
+												<React.Fragment>													
+													{route.authn && this.props.authn == null ? (
+														<Redirect from="current-path" to="/auth" />
+													): null}
+														
+													{!route.hasNoHeader ? (
+														<AppHeader fixed>
+															<Header app={this}/>
+														</AppHeader>
+													) : null}
+
+													<div className="app-body">
+
+														{!route.hasNoSidebar ? (
+															<AppSidebar fixed display="lg">
+																<AppSidebarNav 
+																	navConfig={this.Navigation.getItems()} 
+																	{...this.props} 
+																/>
+																<AppSidebarFooter />
+																<AppSidebarMinimizer />
+															</AppSidebar>
+														) : null}
+
+														<main className="main">
+															
+															{!route.hasNoBreadcrumb ? (
+																<AppBreadcrumb appRoutes={this.Router.Routes}/>
+															) : null}
+															<route.component {...props} {...route.props} />
```

Comment:
If you want to pass the application object, I would go with passing it explicitly:
`{app: this}`

Reply:
but you can keep the route.props, might get useful (y)